cli/sawtooth_cli/network_command/peers.py

```diff
@@ -0,0 +1,150 @@
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import os
+import json
+
+from sawtooth_cli.network_command.parent_parsers import base_multinode_parser
+from sawtooth_cli.network_command.parent_parsers import split_comma_append_args
+from sawtooth_cli.network_command.parent_parsers import make_rest_apis
+
+from sawtooth_cli.exceptions import CliException
+
+
+DOT_FILE = 'peers.dot'
+
+
+def add_peers_parser(subparsers, parent_parser):
+    help_text = 'Shows the peering arrangment of a network'
+
+    parser = subparsers.add_parser(
+        'peers',
+        help=help_text,
+        description='{}.'.format(help_text))
+
+    peers_parsers = parser.add_subparsers(
+        title='subcommands',
+        dest='peers_command')
+    peers_parsers.required = True
+
+    _add_list_parser(peers_parsers, parent_parser)
+    _add_graph_parser(peers_parsers, parent_parser)
+
+
+def _add_list_parser(parser, parent_parser):
+    help_text = 'Lists peers for validators with given URLs'
+
+    list_parser = parser.add_parser(
+        'list',
+        help=help_text,
+        description='{}.'.format(help_text),
+        parents=[parent_parser, base_multinode_parser()])
+
+    list_parser.add_argument(
+        'urls',
+        nargs=1,
+        help='the URLs of the validators to be queried',
+        type=str)
+
+    list_parser.add_argument(
+        '--pretty', '-p',
+        action='store_true',
+        help='Pretty-print the results')
+
+
+def do_peers(args):
+    if args.peers_command == 'list':
+        _do_peers_list(args)
+    elif args.peers_command == 'graph':
+        _do_peers_graph(args)
+    else:
+        raise CliException('Invalid command: {}'.format(args.subcommand))
+
+
+def _do_peers_list(args):
+    urls = split_comma_append_args(args.urls)
+    users = split_comma_append_args(args.users)
+    clients = make_rest_apis(urls, users)
+
+    print(
+        json.dumps(
+            _get_peer_endpoints(clients),
+            sort_keys=True,
+            indent=(4 if args.pretty else 0)
+        )
+    )
+
+
+def _get_peer_endpoints(clients):
+    statuses = [client.get_status() for client in clients]
+
+    return {
+        status['endpoint']: [
+            peer['endpoint']
+            for peer in status['peers']
+        ]
+        for status in statuses
+    }
+
+
+def _add_graph_parser(parser, parent_parser):
+    help_text = "Generates a file to graph a network's peering arrangement"
+
+    graph_parser = parser.add_parser(
+        'graph',
+        help=help_text,
+        description='{}.'.format(help_text),
+        parents=[parent_parser, base_multinode_parser()])
+
+    graph_parser.add_argument(
+        '--output-file',
+        help='The name of the dot file to be produced (defaults to peers.dot)')
+
+    graph_parser.add_argument(
+        '--output-dir',
+        help='The directory in which the dot file is created')
```

Comment:
Typically output is a single flag, see `cli/sawtooth_cli/admin_command/genesis.py` L43

Reply:
`sawtooth keygen` separates the file name and the directory (https://sawtooth.hyperledger.org/docs/core/releases/latest/cli/sawtooth.html#sawtooth-keygen). We can do it either way, but separating them seems more flexible to me.

Reply:
Actually `keygen` does something slightly different, so I'll just go with `--output`.